great_tables/_utils_render_html.py

```diff
@@ -341,50 +317,31 @@ def create_columns_component_h(data: GTData) -> str:
             table_col_headings = tags.tr(level_1_spanners, class_="gt_col_headings gt_spanner_row")
 
     if _get_spanners_matrix_height(data=data) > 2:
```

Comment:
move to separate PR?
<img width="828" alt="Screenshot 2024-05-14 at 15 16 24" src="https://github.com/posit-dev/great-tables/assets/3105306/107ef36c-9cf0-47f0-84d7-4e06af14517d">
<img width="483" alt="Screenshot 2024-05-14 at 15 16 17" src="https://github.com/posit-dev/great-tables/assets/3105306/712ce870-901d-4eaf-8a15-d42c171cfb4d">

Reply:
I would definitely punt this piece to a new PR (or I can take it up). I just tested and spanners above 1 extra level (e.g. a spanner over columns) don't work currently...

```python
from great_tables import GT, exibble

GT(exibble).tab_spanner("A", ["num", "char"]).tab_spanner("B", ["fctr"]).tab_spanner("C", ["A", "B"])
```

Realistically, this code should be raising a NotImplementedError afaict

Reply:
Ah wait, I'm realizing you fixed the spanner implemention! For sure feel free to punt to a separate PR, but realistically you're the hero we all need so if you to keep it in here happy to test / validate in this PR (if more convenient)

Reply:
It's not quite correct, im going to breakout and finish in a separate PR

Reply:
https://github.com/posit-dev/great-tables/pull/345